contracts/vaults/common/VaultRewarderLib.sol

```diff
@@ -132,6 +135,51 @@ contract VaultRewarderLib is IVaultRewarder, ReentrancyGuard {
         emit VaultRewardUpdate(rewardToken, emissionRatePerYear, endTime);
     }
 
+    function _withdrawFromPreviousRewardPool(IERC20 poolToken, RewardPoolStorage memory r) internal {
+        if (r.rewardPool == address(0)) return;
+
+        // First, withdraw from existing pool and clear approval
+        uint256 boosterBalance = IERC20(address(r.rewardPool)).balanceOf(address(this));
+        if (r.poolType == RewardPoolType.AURA) {
+            require(IAuraRewardPool(r.rewardPool).withdrawAndUnwrap(boosterBalance, true));
+        } else if (r.poolType == RewardPoolType.CONVEX_MAINNET) {
+            require(IConvexRewardPool(r.rewardPool).withdrawAndUnwrap(boosterBalance, true));
+        } else if (r.poolType == RewardPoolType.CONVEX_ARBITRUM) {
+            require(IConvexRewardPoolArbitrum(r.rewardPool).withdraw(boosterBalance, true));
+        }
+
+        // Clear approvals on the old pool.
+        poolToken.checkApprove(address(r.rewardPool), 0);
+    }
+
+    function migrateRewardPool(IERC20 poolToken, RewardPoolStorage memory newRewardPool) external {
+        require(msg.sender == Deployments.NOTIONAL.owner());
+        RewardPoolStorage memory r = VaultStorage.getRewardPoolStorage();
+
+        _withdrawFromPreviousRewardPool(poolToken, r);
```

Comment:
Should we call claimRewardTokens at the beginning and end of this function? Would make sure that we're not leaving any incentives on the table from the last pool and that the lastClaimTimestamp on the new pool is up to date.